src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
```

Comment:
PullProofAI

The `getUserData` method is vulnerable to SQL injection due to string interpolation. Use parameterized queries instead.
```diff
  public getUserData(userId: string): string {
-12:     const query = `SELECT * FROM users WHERE id = '${userId}'`;
+12:     const query = 'SELECT * FROM users WHERE id = $1';
+13:     return this.executeQuery(query, [userId]);
```



<!-- This is an auto-generated comment by PullProofAI -->